orderbook/src/orderbook.rs

```diff
@@ -85,6 +89,52 @@ impl OrderBook {
         orders.retain(|_, order| has_future_valid_to(now_in_epoch_seconds, &order.order_creation));
     }
 
+    async fn remove_settled_orders(&self, settlement_contract: &GPv2Settlement) {
+        let uids_to_be_removed = self.get_uids_of_settled_orders(settlement_contract).await;
+        info!(
+            "removing following settled orders from orderbook: {:?}",
+            uids_to_be_removed
+        );
+        let mut orders = self.orders.write().await;
+        orders.retain(|uid, _| uids_to_be_removed.contains(uid));
+    }
+
+    async fn get_uids_of_settled_orders(
+        &self,
+        settlement_contract: &GPv2Settlement,
+    ) -> Vec<OrderUid> {
+        let orders = self.orders.read().await;
+        let uid_futures = orders.iter().map(|(uid, _)| async move {
+            (
+                *uid,
+                self.has_not_yet_been_settled(*uid, settlement_contract)
+                    .await,
+            )
+        });
+        let uid_pairs: Vec<(OrderUid, bool)> = join_all(uid_futures).await;
+        uid_pairs
+            .iter()
+            .filter(|(_, not_yet_settled)| *not_yet_settled)
+            .map(|(uid, _)| *uid)
```

Comment:
Ubernit: consider using [filter_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) directly?

Reply:
I looked into it, but I did not find an example on how to filter map a touple on the first argument only...